lib/tcp.js

```diff
@@ -30,8 +28,10 @@ module.exports = {
       onData(msg, parser, debug, items);
     });
     client[items].on('close', () => {
-      debug('TCP connection closed');
-      onClose(host, port, parser, debug, items);
+      setTimeout(() => {
+        debug('TCP connection closed');
```

Comment:
debug message fires 10 seconds after close

Reply:
Latest commit has refactorings to functions and this issue is correct in that